starboard/android/shared/audio_decoder.cc

```diff
@@ -94,9 +94,10 @@ void AudioDecoder::Initialize(const OutputCB& output_cb,
 
   output_cb_ = output_cb;
   error_cb_ = error_cb;
-
-  media_decoder_->Initialize(
-      std::bind(&AudioDecoder::ReportError, this, _1, _2));
+  if (media_decoder_) {
```

Comment:
There is a risk that `is_valid` in the future may use an alternate criteria for `validity`. I want this code to explicitly never dereference a null pointer. Since in theory there can be an edge case where `InitializeCodec` returns with it being null, or `Reset` may have been called.